apps/prairielearn/src/ee/lib/lti13.sql

```diff
@@ -94,3 +94,51 @@ DELETE FROM lti13_assessments
 WHERE
   lti13_course_instance_id = $lti13_course_instance_id
   AND assessment_id = $assessment_id;
+
+-- BLOCK select_assessment_instances_for_scores
+SELECT
+  ai.*,
+  CASE
+    WHEN a.group_work
+    AND ai.group_id IS NOT NULL THEN (
+      SELECT
+        jsonb_agg(
+          to_jsonb(users) || jsonb_build_object('lti13_sub', lti13_users.sub)
+        )
+      FROM
+        group_users
+        JOIN users ON (group_users.user_id = users.user_id)
+        LEFT JOIN lti13_users ON (lti13_users.user_id = group_users.user_id)
+      WHERE
+        group_users.group_id = ai.group_id
+    )
+    ELSE jsonb_build_array(
+      to_jsonb(u) || jsonb_build_object('lti13_sub', lu.sub)
+    )
+  END AS users
+FROM
+  assessment_instances AS ai
+  JOIN assessments AS a ON (a.id = ai.assessment_id)
+  LEFT JOIN users AS u ON (u.user_id = ai.user_id)
+  LEFT JOIN lti13_users AS lu ON (lu.user_id = ai.user_id)
+WHERE
+  ai.assessment_id = $assessment_id
+  AND ai.score_perc IS NOT NULL
+  AND ai.date IS NOT NULL;
+
+-- BLOCK select_assessment_with_lti13_course_instance_id
+SELECT
+  a.*,
+  la.lineitem_id_url AS lti13_lineitem_id_url,
```

Comment:
Because of the `LEFT JOIN`, this might be `NULL`, but the types you use when querying this have `lti13_lineitem_id_url: z.string()` (meaning it's required). We generally want our types and queries to agree.

Reply:
I was using this same select in two different places in lti.ts with different expectations (and types). I broke that out to two different queries and fixed the agreement between this type and query.

Reply:
Addressed in a0c9223d3